src/OutputWriters/checkpointer.jl

```diff
@@ -213,8 +213,12 @@ function set!(model, filepath::AbstractString)
                 parent_data = file["$name/data"]
                 model_field = model_fields[name]
                 copyto!(model_field.data.parent, parent_data)
-            catch
-                @warn "Could not retore $name from checkpoint."
+            catch err
+                if err isa KeyError
```

Comment:
Why KeyError? Maybe add a comment?

Reply:
That's the expected error when a variable isn't present in the checkpoint. It's possible to have other errors though, so I think should `rethrow()` the error if something else goes wrong

Reply:
In any case your comment made me realize the way it was written probably isn't best practice, so I re-wrote the code using an if-else statement instead of a try-catch. The new code is shorter and (I think) clearer. What do you think?

Reply:
I think it's easier to understand.

There's the concept of "Easier to Ask for Forgiveness than Permission" which claims that try/catch is better style esp for dynamic languages.

I think our project and code is unique. So while there are places that so-called "best practices" apply, there are also places where we should do what works best for us.

Reply:
The downside of "look before you leap" is when we have many error conditions which causes the if-statements to pile up. Here we have just one condition so I think explicit writing the condition is better and more readable.